docs/source/miscellaneous/doc/adr/0016-use-tickets-encoding-for-transactions.md

```diff
@@ -253,14 +255,256 @@ performance would be poor for services with many nodes.
+predicate.
+
+Thus, in Atlas when we try and perform a get of a collection of cells (which are row-column pairs), we first
+group the pairs by column and then, in parallel, dispatch requests to Cassandra for each row that is relevant.
+For example, if one's cells were ``(A, 1), (A, 2), (B, 1), (C, 2), (C, 3), (D, 3)``, then Atlas would send three
+requests:
+
+- column ``1`` and keys ``[A, B]``
+- column ``2`` and keys ``[A, C]``
+- column ``3`` and keys ``[C, D]``
+
+Note that in practice, the requests we make have to use a range predicate on Cassandra, because cells don't include
+timestamps, and the latest timestamp at which our cell existed isn't something we know a priori.
+
+#### Multiget Multislice
+
+The above model does not work well for transactions2. Transactions2 cells end up being distributed reasonably evenly
+among the columns 0 through PQ / NP (which, in our case, is 312,500). Thus, when attempting to determine whether some
+Atlas values had been committed, we will perform many requests in parallel. These requests will end up using many
+resources from the Cassandra connection pool; they also incur a lot of overhead in terms of scheduling and network I/O.
+
+We want to be able to batch these calls together. To do this, we added another endpoint to the Thrift interface that
+Palantir's fork of Cassandra provides:
+
+```
+struct KeyPredicate {
+    1: optional binary key,
+    2: optional SlicePredicate predicate,
+}
+
+map<binary,list<list<ColumnOrSuperColumn>>> multiget_multislice(1:required list<KeyPredicate> request,
+                                                                2:required ColumnParent column_parent,
+                                                                3:required ConsistencyLevel consistency_level=ConsistencyLevel.ONE)
+  throws (1:InvalidRequestException ire, 2:UnavailableException ue, 3:TimedOutException te)
+```
+
+Implementing this endpoint on the Cassandra side was not too difficult. It may seem a little wasteful in that this may
+require keys and predicates to be specified more than once, but for transactions2 these would likely be mostly
+distinct.
+
+While this improved performance, we still faced significant regressions relative to the v1 cell loader. We determined
+that this was because Cassandra has a worker pool for loading values to satisfy a ``ReadCommand``, but the calling
+thread in requests is also allowed to participate. Thus, creating large batches would turn out to likely be detrimental
+to read performance, even when the Cassandra nodes are actually able to handle higher concurrency safely.
+
+#### Selective Batching
+
+We thus settled on a compromise between sending large singular requests and inundating Cassandra with smaller ones;
+unlike in the original ``CellLoader``, we make the batch parameters configurable. There are two parameters:
+
+- cross column load batch limit (CC); we may combine requests for different columns in one call to the DB, but merged
```

Comment:
I think abbreviations that represent symbols should be wrapped in backticks.

Reply:
Sure, added CC and SQ. In general it does make sense to differentiate the symbols in some way, though I tend to be a bit hesitant to do that as the backticks can make the raw form hard to read